examples/fluids/src/setuplibceed.c

```diff
@@ -163,7 +163,16 @@ PetscErrorCode SetupLibceed(Ceed ceed, CeedData ceed_data, DM dm, User user, App
   // Set up CEED objects for the interior domain (volume)
   // *****************************************************************************
   const PetscInt num_comp_q = 5;
-  const CeedInt  dim = problem->dim, num_comp_x = problem->dim, q_data_size_vol = problem->q_data_size_vol, jac_data_size_vol = num_comp_q + 6 + 3;
+  const CeedInt  dim = problem->dim, num_comp_x = problem->dim, q_data_size_vol = problem->q_data_size_vol;
+  CeedInt        jac_data_size_vol = num_comp_q + 6 + 3;
+
+  if (problem->apply_vol_ifunction.qfunction) {
+    NewtonianIdealGasContext gas;
+    PetscCallCeed(ceed, CeedQFunctionContextGetDataRead(problem->apply_vol_ifunction.qfunction_context, CEED_MEM_HOST, &gas));
+    jac_data_size_vol += (gas->idl_enable ? 1 : 0);
```

Comment:
```suggestion
  CeedInt        jac_data_size_vol = num_comp_q + 6 + 3 + (gas->idl_enable ? 1 : 0);

  if (problem->apply_vol_ifunction.qfunction) {
    NewtonianIdealGasContext gas;
    PetscCallCeed(ceed, CeedQFunctionContextGetDataRead(problem->apply_vol_ifunction.qfunction_context, CEED_MEM_HOST, &gas));
```

Reply:
Ope, you're right. Didn't noticed that `idl_enable` was coming from the QF context.